### DISCUSSION TITLE
Receiving a lot of SPAM-E Mails

### DISCUSSION THREAD

**Discussion by Gill-Bates** [modoboa/modoboa#1984] [Help]
On a freshly installed Modoba instance I have to realize that - despite all protection measures - I keep receiving 5-10 SPAM mails per day. Do I have to give the system time to train the filters or do I have to adjust certain parameters in the settings?

**Comment by neonknight** [comment]
without knowing the exact type of spam your receive or seeing some samples it is impossible to give any hints on how to improve your filter.

building a spam filter is certainly not trivial and the modoboa setup (like every free 'host yourself' filter) is providing a basic filter that works reasonably well for most and focuses on not having false positives. If you want more you'll need to dive deep into this matter, which means:
- configure more restrictions and rules in postfix
- add additional RBLs to either postfix or spamassassin
- add or enable additional spamassassin plugins
- add additional rule sets to spamassassin and/or tweak scores
- improve bayes training (first by learning missed spam, but it can be partially automated by seeding and configuring spam trap addresses)
- add commercial antivirus and/or third party rules for clamav

as all of this requires a lot of work, experience and sometimes even money, it may be reasonable to just look for a professional/commercial spam filter operator in cases the spam load is not acceptable.

**Comment by divo1** [comment]
I have similar problem. Can you add some links for this points? O can you describe some of this? 
"improve bayes training" is by setting mails as spam. Right? I have ~500 spam with similar subjects and still getting these to inbox (of course I have more spam with other subjects). Every one of these I "mark as spam". 
Maybe you can share with some links or tips?

**Comment by Gill-Bates** [comment]
I use a catch-all domain and receive a lot of spam mails.  They always come at night at 2 o'clock in the form of a wave of 15-20 mails with sexual content.  I delete them daily and mark them as spam.  But it seems that there is no learning filter here.

An anti-spam strategy should always include several aspects.  For example, SPF check, Spamhouse lookups or Spam Assasian score.  Graylisting is outdated and user unfriendly.  I claim that you can get your spam under control very well with free products without having to switch to commercial products immediately.

**Comment by divo1** [comment]
I resolve this by using sieve filters, but still I was need to add this emails every week. I thought that here I can do this in better way. 
Hmm... I was thinking about graylisting. Maybe this wyould help.

**Comment by Gill-Bates** [comment]
graylisting is out!  Imagine a password reset and you have to wait 15 minutes for the email.

**Comment by MrGeneration** [comment]
> graylisting is out! Imagine a password reset and you have to wait 15 minutes for the email.

That's totally fine to me. It does reduce spamers that only sendout the mail once (not via MTA).
Yes, they do still exist.

A negative effect of that is that you will not receive mails by e.g. aliexpress, because they seem to do the same. :-D

**Comment by tonioo** [comment]
@Gill-Bates Can I close this issue?

**Comment by blu-IT** [comment]
I have this problem too, that al lot of spam passes the spam-filter. Only a minority of the spam messages are recognized and blocked.

I installed modoboa with the install script and activated spamassassin / amavis ... but it seems to me, that there are no further options to fine tune spamassassion or train the system.

**Comment by blu-IT** [comment]
I am trying no to train spamassassin with "sa-learn", but honestly I did not find a process with spam running on the system. Only amavis / amavisd-new.
The manual / how to for training spamassassin with spam data says, that you have to run "sa-learn" as the user who is running spamc on the system. I'm not sure now... sa-learn does not work as amavis. Trying as root now...
